iModelCore/Units/PrivateAPI/Formatting/FormattingParsing.h

```diff
@@ -144,6 +144,11 @@ struct FormatParsingSet
     //! Returns error codes when FUS does not match the expression.
     //! The input expression signature code mus be provided by the caller
     BEU::Quantity ComposeColonizedQuantity(Formatting::FormatSpecialCodes cod, FormatCP fusP = nullptr);
+    BEU::Quantity ParseAndProcessTokens(Formatting::FormatSpecialCodes cod, FormatCP fusp, BEU::UnitCP inputUnit);
+    // parsing helper methods
+    BEU::Quantity ParseBearingFormat(FormatProblemCode* probCode, FormatCP fusP, BEU::UnitCP unit);
+    BEU::Quantity ParseRatioFormat(FormatProblemCode* probCode,FormatCP fusP, BEU::UnitCP unit);
+    BEU::Quantity UpdateAndSetProblemCode(FormatProblemCode code, FormatProblemCode* probCode);

```

Comment:
This is a very strange method. The second parameter returns a pointer to the first parameter, and the method always returns an empty Quantity when we probably have a quantity object with a more precise error value available. I don't like this method, and I don't mind the bit of redundancy which it fixes. Let's just replace the calls, for example

return UpdateAndSetProblemCode(FormatProblemCode::QT_MathematicOperationFoundButIsNotAllowed, probCode);

By the full bloated version, like

probCode = FormatProblemCode::QT_MathematicOperationFoundButIsNotAllowed;
return converted; //quantity has error assigned

And then at the outmost caller (GetQuantity()) we can check if a problemCode has been set to the pointer and if so, call m_problem.UpdateProblemCode(code);